CMakeLists.txt

```diff
@@ -248,6 +248,10 @@ if (COMPILER_SUPPORTS_AVX2)
   add_compile_options(${COMPILER_AVX2_FLAG})
 endif()
 
+# Formatting and documentation
+include(Doxygen)
+include(Format)
+
```

Comment:
These would be better placed with all the other inclusions closer to the top the file.

(If there's some reason they need to happen after the `project` statement, we should have a second block of inclusions immediately after.)